lib/ansible/plugins/inventory/constructed.py

```diff
@@ -116,6 +116,11 @@ def parse(self, inventory, loader, path, cache=False):
                 # create composite vars
                 self._set_composite_vars(data.get('compose'), hostvars, host, strict=strict)
 
+                # refetch host vars in case new ones have been created above
+                hostvars = inventory.hosts[host].get_vars()
+                if host in self._cache:  # adds facts if cache is active
+                    hostvars = combine_vars(hostvars, self._cache[host])
```

Comment:
this seems to do the reverse of what you want

Reply:
I tested this locally and it worked as I wanted/expected. _set_composite_vars sets the host vars in the inventory then this retrieves them from the inventory.

Reply:
the order is incorrect, it probably 'works' if you don't have duplicate vars in  inventories as the set_variable is overriding any cached vars you would have yet you do the reverse override here.

Reply:
tempted to just say, use hostvars ... as they would be the same as the cache + new combined vars already

Reply:
I see. Using the same variable name just seems like asking for trouble to me :) So just to be clear you're suggesting just update the local hostvars at the same time we update the inventory? I'd be happy with that.

Reply:
he, sadly most people do use duplicate names across hosts/groups

no, just do line #120 and remove 121-122 as they become redundant